libs/framework/include/celix/Utils.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+
+#include <string>
+#include <string.h>
+#include <iostream>
+
+namespace celix {
+    /**
+     * @brief Returns the deferred type name for the template I
+     */
+    template<typename INTERFACE_TYPENAME>
+    std::string typeName() {
+        std::string result;
+
+#ifdef __GXX_RTTI
```

Comment:
I think this would work for clang as well, if one were to use something like https://gist.github.com/brimston3/2be168bb423c83b0f469c0be56e66d31

Reply:
Yes, this is a good improvement. updated